File: Day-2/day2.hs
main = do
    fileData <- readFile "input.txt"
    let plays = map play (lines fileData)
                where play s = let (s1:s2:[]) = words s in (s1 !! 0, s2 !! 0)

    print $ "Solution to part 1: " ++ show (sum $ map gameOutcomeScores1 plays)
    print $ "Solution to part 2: " ++ show (sum $ map gameOutcomeScores2 plays)

gameOutcomeScores1 :: (Char, Char) -> Int
gameOutcomeScores1 ('A','X') = 4
gameOutcomeScores1 ('B','Y') = 5
gameOutcomeScores1 ('C','Z') = 6
gameOutcomeScores1 ('B','X') = 1
gameOutcomeScores1 ('C','X') = 7
gameOutcomeScores1 ('A','Y') = 8
gameOutcomeScores1 ('C','Y') = 2
gameOutcomeScores1 ('A','Z') = 3
gameOutcomeScores1 ('B','Z') = 9

gameOutcomeScores2 :: (Char, Char) -> Int
gameOutcomeScores2 ('A','X') = 3
gameOutcomeScores2 ('B','Y') = 5
gameOutcomeScores2 ('C','Z') = 7
gameOutcomeScores2 ('B','X') = 1
gameOutcomeScores2 ('C','X') = 2
gameOutcomeScores2 ('A','Y') = 4
gameOutcomeScores2 ('C','Y') = 6
gameOutcomeScores2 ('A','Z') = 8
gameOutcomeScores2 ('B','Z') = 9

File: Day-2/day2.py
gameOutcomeScores1 = {
    "X - A": 4, 
    "Y - B": 5, 
    "Z - C": 6, 
    "X - B": 1, 
    "X - C": 7, 
    "Y - A": 8, 
    "Y - C": 2, 
    "Z - A": 3, 
    "Z - B": 9
}

gameOutcomeScores2 = {
    "X - A": 3, 
    "Y - B": 5, 
    "Z - C": 7, 
    "X - B": 1, 
    "X - C": 2, 
    "Y - A": 4, 
    "Y - C": 6, 
    "Z - A": 8, 
    "Z - B": 9
}

scorePartA = 0
scorePartB = 0
file = open("Day-2/test.txt", "r")

for line in file:
    curr = line.split(" ")
    play = curr[1].strip() + " - " + curr[0].strip()
    scorePartA += gameOutcomeScores1.get(play)
    scorePartB += gameOutcomeScores2.get(play)

file.close()

print("Solution to part 1: ", scorePartA) 
print("Solution to part 2: ", scorePartB)